app/app.go

```diff
@@ -203,35 +201,39 @@ type UmeeApp struct {
 	legacyAmino       *codec.LegacyAmino
 	appCodec          codec.Codec
 	interfaceRegistry types.InterfaceRegistry
+	txConfig          client.TxConfig
 
 	invCheckPeriod uint
 
 	// keys to access the substores
-	keys    map[string]*sdk.KVStoreKey
-	tkeys   map[string]*sdk.TransientStoreKey
-	memKeys map[string]*sdk.MemoryStoreKey
+	keys    map[string]*storetypes.KVStoreKey
+	tkeys   map[string]*storetypes.TransientStoreKey
+	memKeys map[string]*storetypes.MemoryStoreKey
 
 	// keepers
 	AccountKeeper    authkeeper.AccountKeeper
 	BankKeeper       bankkeeper.Keeper
 	CapabilityKeeper *capabilitykeeper.Keeper
-	StakingKeeper    stakingkeeper.Keeper
+	StakingKeeper    *stakingkeeper.Keeper
```

Comment:
This `UmeeApp` field may be responsible for the current error in `make test-sim-non-determinism`:

> --- FAIL: TestAppStateDeterminism (0.03s)
> panic: interface conversion: types.StakingKeeper is *keeper.Keeper, not keeper.Keeper [recovered]
> 	panic: interface conversion: types.StakingKeeper is *keeper.Keeper, not keeper.Keeper

The stack trace included `github.com/cosmos/cosmos-sdk/x/distribution/simulation.WeightedOperation`

We might need to revert or otherwise work around this

Reply:
This is easy to fix, we just need to update it in a test.

Reply:
Btw, reverting this does allow sims to run, but then getting a panic from x/group: 
```
error on block  1/500, operation (18/291) from x/group:
        recovered: value is nil
...
        : panic [umee-network/cosmos-sdk@v0.46.0-umee.0.20220812010629-4d5bb2e3f73c/baseapp/recovery.go:69]
        Comment: unable to deliver tx
```

Reply:
We need pointer there. distribution and staking slashing module should accept interface type, not a concrete type. 
I fixed it in the module constructor, and also submitted patch for the SDK to depend on interface.